src/providers/yugen.rs

```diff
@@ -0,0 +1,72 @@
+use regex::Regex;
+use reqwest::{Client, Response};
+use reqwest_middleware::ClientWithMiddleware;
+use scraper::{Html, Selector, element_ref::Select};
+
+use crate::types::{AnimeEpisode, SearchResult, StreamLink};
+
+const host: &str = "https://yugen.to";
+
+pub async fn search(args: (&ClientWithMiddleware, &str)) -> Option<Vec<SearchResult>> {
+    let (client, query) = args;
+    let res: String = client
+        .get(format!("{}/search/?q={}", host, query))
```

Comment:
Please either url_encode the param or use proper params (via a HashMap or whatever reqwest uses)